File: day16-Sorting/Solution.py
N = input()
li = sorted(map(int, raw_input().split()))
minDiff = abs(li[0] - li[1])

for i in xrange(N-1):
    diff = abs(li[i] - li[i+1])
    if diff < minDiff:
        minDiff = diff
        
for i in xrange(N-1):
    diff = abs(li[i] - li[i+1])
    if diff == minDiff:
        print li[i], li[i+1],

File: day16-Sorting/Solution.java
import java.util.*;

public class Solution 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int arr[]= new int[N];
        
        for (int i = 0; i < N; i++) arr[i] = sc.nextInt();
        Arrays.sort(arr);
        int minDiff = Math.abs(arr[0] - arr[1]);
        for (int i = 0; i < N -1; i++)
        {
            int diff = Math.abs(arr[i] - arr[i+1]);
            minDiff = diff < minDiff ? diff:minDiff;
        }
        
        for (int i = 0; i < N -1; i++)
        {
            int diff = Math.abs(arr[i] - arr[i+1]);
            if (diff == minDiff) System.out.printf("%d %d ",arr[i],arr[i+1]);
        }
    }
}